### ISSUE TITLE
Windows system installer creates invalid Windows System PATH entry of "C".

### ISSUE THREAD

**Issue by johnl2323** [doxygen/doxygen#10614]
**Describe the bug**
When installing Windows doxygen using the Windows System Installer (Innosetup .exe installer), the installer creates a new System PATH entry which is just a "C".   This is an invalid PATH entry.

I assume it was trying to add C:\Program Files\doxygen\bin to the System PATH. 

**Expected behavior**

I would have expected it to create a System PATH entry of where doxygen was installed so "C:\Program Files\doxygen\bin".

**Screenshots**
Screenshot from Windows Edit Environment Variables window showing the base "C" entry added by doxygen install.

![image](https://github.com/doxygen/doxygen/assets/11217469/49d0dec2-2141-4db8-a6df-c753bd23a5e8)

**To Reproduce**

1. Download the Windows 1.10.0 system installer (doxygen-1.10.0-setup.exe) and install doxygen
2. After installation goto the Windows Edit Environment Variables control panel and edit the System PATH env var.
3. You will notice the an entry of just "C" was added to the PATH

**Version**
Windows 1.9.3 and 1.10.0 system installers exhibited the issue:

doxygen-1.9.3-setup.exe
doxygen-1.10.0-setup.exe

**Comment by albert-github** [comment]
I never noticed before, but I see in my Windows 10 environment when giving the command path: `path` the strange sequence:
```
D;D;D;D;D;D;D;D;D;D;D;D;D;D;D;D;D;D;D;
```
a bit later in my path I see `d:\Program Files\doxygen\bin` so this might be the reason I never noticed.


You write:
> Windows 1.9.3 and 1.10.0 system installers exhibited the issue

you probably didn't test other versions, but I assume that it might be present in the intermediate versions as well and maybe also in some older versions.

**Comment by johnl2323** [comment]
I didn't test the intermediate versions.   We are currently using 1.9.3 in general use and I saw the issue.  So I figured I would test the latest release to see if it still is an issue in 1.10.0 and it was.

Also I tried looking for the source code to create the Innosetup windows package, but didn't find it...

**Comment by doxygen** [comment]
This is the code section of the `doxygen.iss` script. It hasn't been changed for many years, so it probably needs some updating. Let me know if you spot the problem.

```.iss
[code]

program Setup;

function AddPathToAutoExecBat(PathToAdd: String): Boolean;
var
   DLLHandle: LongInt;
   ErrorCode: Integer;
   StrPointer: LongInt;
   ProcResult: LongInt;
begin
   Result := False;

   DLLHandle := LoadDLL(ExpandConstant('{tmp}\isutils.dll'), ErrorCode);

   if DLLHandle > 31 then begin
      StrPointer := CastStringToInteger(PathToAdd);
      Result := CallDLLProc(DLLHandle, 'AddPathToAutoExec', StrPointer, 0, ProcResult);
      Result := FreeDLL(DLLHandle);
      //if not ProcResult = 1 then
      //   Result := False;
   end else begin
      MsgBox('Setup is unable to load isutils.dll', mbError, MB_OK);
      Result := False;
   end;
end;

function AddPathToRegistry(PathToAdd: String): Boolean;
var
   DLLHandle: LongInt;
   ErrorCode: Integer;
   StrPointer: LongInt;
   ProcResult: LongInt;
begin
   Result := False;

   if not IsAdmin() then begin
      MsgBox('Setup cannot update PATH environment variable in the system registry.' #13#13 'Login as an Administrator and run this Setup again', mbError, MB
   end else
   begin

      DLLHandle := LoadDLL(ExpandConstant('{tmp}\isutils.dll'), ErrorCode);

      if DLLHandle > 31 then begin
         StrPointer := CastStringToInteger(PathToAdd);
         Result := CallDLLProc(DLLHandle, 'AddPathToRegistry', StrPointer, 0, ProcResult);
         Result := FreeDLL(DLLHandle);
         //if not ProcResult = 1 then
         //   Result := False;
      end else begin
         MsgBox('Setup is unable to load isutils.dll', mbError, MB_OK);
         Result := False;
      end;
   end;
end;

procedure CurStepChanged(CurStep: TSetupStep);
var
  PathVar: String;
  ResVal: Boolean;
begin
  if CurStep = ssPostInstall then begin

     //get current path info
     ResVal := RegQueryStringValue(HKEY_LOCAL_MACHINE, 'SYSTEM\CurrentControlSet\Control\Session Manager\Environment', 'Path', PathVar);
     if not ResVal then
        PathVar := GetEnv ('PATH');

    //if the application path doesn't exist, add it
    if Pos(Lowercase(ExpandConstant('{app}\bin')), Lowercase(PathVar)) = 0 then begin
       if UsingWinNT() then
          ResVal := AddPathToRegistry(ExpandConstant('{app}\bin'))
       else
          ResVal := AddPathToAutoExecBat(ExpandConstant('"{app}\bin"'));

       if not ResVal then
          MsgBox('Setup is unable to update PATH environment variable!', mbError, MB_OK);
    end;

  end;
end;

procedure CurUninstallStepChanged(CurUninstallStep: TUninstallStep);
var
  BinDir, Path: String;
begin
  if (CurUninstallStep = usPostUninstall)
     and (RegQueryStringValue(HKEY_CURRENT_USER, 'Environment', 'PATH', Path)) then
  begin
    BinDir := ExpandConstant('{app}\bin');
    if Pos(LowerCase(BinDir) + ';', Lowercase(Path)) <> 0 then
    begin
      StringChange(Path, BinDir + ';', '');
      RegWriteStringValue(HKEY_CURRENT_USER, 'Environment', 'PATH', Path);
    end;
  end;
end;

begin
end.
```

**Comment by albert-github** [comment]
@doxygen 

I don't see a problem in the given code, the strange thing is also that it is just the drive letter and not the drive letter plus colon (as that might have indicated a "backslash" problem).

Would it be possible to have a "debug" installer i.e. something where we can get more output during the install process?
I already saw the `/log` / `/log=filename` possibility of Inno Setup but in the generated log file I didn't find anything related to the setting of `HKEY_CURRENT_USER` / Environment etc.

**Comment by doxygen** [comment]
I've changed the code in the innosetup script based on [1](https://stackoverflow.com/a/46609047/784672), you can find a test executable [here](https://www.doxygen.nl/files/doxygen-1.11.0-test-setup.exe). Let me know if it works for you.

**Comment by albert-github** [comment]
I did a number of tests and it looks to me that the problem in the mentioned test installation is solved.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
